### PR TITLE
[IMP]website_livechat: Convert into new API

### PR DESCRIPTION
**Pull request by gan-odoo**
Task : https://www.odoo.com/web#id=12125&view_type=form&model=project.task&menu_id=3942&action=327
Pad : https://pad.odoo.com/p/r.f7qWOImZhcJ42enS

### CODE REVIEW

File: addons/website_livechat/models/im_livechat.py [hunk 1]
```diff
@@ -3,16 +3,15 @@
...
-    # TODO : when mixin in new api.v8, change this !
-    def _website_url(self, cr, uid, ids, field_name, arg, context=None):
-        res = super(im_livechat_channel, self)._website_url(cr, uid, ids, field_name, arg, context=context)
-        for channel in self.browse(cr, uid, ids, context=context):
-            res[channel.id] = "/livechat/channel/%s" % (slug(channel),)
+    @api.multi
+    @api.depends('name')
+    def _website_url(self, name, arg):
+        res = super(ImLivechatChannel, self)._website_url(name, arg)
+        res.update({(livechat.id, '/livechat/channel/%s' % slug(livechat)) for livechat in self})
```
Comment (xmo-odoo) [inline addons/website_livechat/models/im_livechat.py]: Why the `@depends`?
Why a set comprehension inside a call to `dict.update`?

File: addons/website_livechat/models/website.py [hunk 2]
```diff
@@ -18,15 +17,16 @@ class website_config_settings(models.TransientModel):
...
-        qcontext = super(ir_ui_view, self)._prepare_qcontext(cr, uid, context=context)
+    @api.model
+    def _prepare_qcontext(self):
+        qcontext = super(IrUiView, self)._prepare_qcontext()
         if request and getattr(request, 'website_enabled', False):
             if request.website.channel_id:
-                qcontext['website_livechat_url'] = self.pool.get('ir.config_parameter').get_param(cr, SUPERUSER_ID, 'web.base.url')
-                qcontext['website_livechat_dbname'] = cr.dbname
+                qcontext['website_livechat_url'] = self.env['ir.config_parameter'].get_param('web.base.url')
+                qcontext['website_livechat_dbname'] = self._cr.dbname
```
Comment (xmo-odoo) [inline addons/website_livechat/models/website.py]: `Model._cr` should be avoided, they're here for BC reasons. Get the cursor from the env.

File: addons/website_livechat/models/im_livechat.py (hunk 1 above)
Comment (gan-odoo) [inline addons/website_livechat/models/im_livechat.py]: Yes, No need to @depends.
Done!

File: addons/website_livechat/models/website.py (hunk 2 above)
Comment (gan-odoo) [inline addons/website_livechat/models/website.py]: Done!

File: addons/website_livechat/models/im_livechat.py (hunk 1 above)
Comment (xmo-odoo) [inline addons/website_livechat/models/im_livechat.py]: There's still a set comprehension inside the call to update, I don't see why.

Comment (gan-odoo) [inline addons/website_livechat/models/im_livechat.py]: Please check compare. I have push this change.

File: addons/website_livechat/views/website_livechat_view.xml [hunk 3]
```diff
@@ -1,6 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
-    <data>
+<odoo>
 
         <record id="im_livechat_channel_form_view" model="ir.ui.view">
             <field name="name">im_livechat.channel.form.with.website_published</field>
```
Comment (jem-odoo) [inline addons/website_livechat/views/website_livechat_view.xml]: rename it into `im_livechat.channel.form.inherit`

Comment (gan-odoo) [inline addons/website_livechat/views/website_livechat_view.xml]: OK Done!

**Comment by hmo-odoo** [comment]
@gan-odoo , can you apply new guidelines 
* make python file for each model
* put data file in data directory.

**Comment by gan-odoo** [comment]
@hmo-odoo , I have changed as per new guidline.
python file splite each model.
data file move in data directory.

File: addons/website_livechat/models/website.py [hunk 4]
```diff
@@ -1,7 +1,5 @@
 # -*- coding: utf-8 -*-
-from openerp.osv import osv
 from openerp import api, fields, models
```
Comment (rgo-odoo) [inline addons/website_livechat/models/website.py]: remove unused import "api"

File: addons/website_livechat/models/ir_ui_view.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+from openerp import api, fields, models
+from openerp.http import request
+
+class IrUiView(models.Model):
```
Comment (rgo-odoo) [inline addons/website_livechat/models/ir_ui_view.py]: this class been removed from master in rev 132e96ac, you should too.

**Comment by rgo-odoo** [comment]
it doesn't make any sense to set default=False for an HTML type field in website_livechat/models/im_livechat.py

File: addons/website_livechat/models/ir_ui_view.py (hunk 5 above)
Comment (gan-odoo) [inline addons/website_livechat/models/ir_ui_view.py]: OK, Removed this.

**Comment by hmo-odoo** [comment]
@mart-e , @gan-odoo . website_livechat module already in new api and this PR has minor changes so I closed this PR bz it's not related migration.

**Pull request closed without merging**